sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -159,6 +161,61 @@ public ShareServiceVersion getServiceVersion() {
         return serviceVersion;
     }
 
+    /**
+     * Gets if the file this client represents exists in the cloud.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.ShareFileAsyncClient.exists}
+     *
+     * @return true if the file exists, false if it doesn't
+     */
+    public Mono<Boolean> exists() {
+        try {
+            return existsWithResponse().flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Gets if the file this client represents exists in the cloud.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.share.ShareFileAsyncClient.existsWithResponse}
+     *
+     * @return true if the file exists, false if it doesn't
+     */
+    public Mono<Response<Boolean>> existsWithResponse() {
+        try {
+            return withContext(this::existsWithResponse);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Gets if the file this client represents exists in the cloud.
+     *
+     * @return true if the file exists, false if it doesn't
+     */
+    Mono<Response<Boolean>> existsWithResponse(Context context) {
+        return this.getPropertiesWithResponse(null, context)
+            .map(cp -> (Response<Boolean>) new SimpleResponse<>(cp, true))
+            .onErrorResume(t ->
+                    (t instanceof ShareStorageException && ((ShareStorageException) t).getStatusCode() == 404)
+                || (t instanceof HttpResponseException && ((HttpResponseException) t).getResponse().getStatusCode()
+                        == 404),
```

Comment:
Does this method return `HttpResponseException` in some cases? Just noticed this check differs from the others.

Reply:
I think our serialization of exceptions happens depending on the XML returned by the service. The file `getProperties` request is a HEAD request so it doesnt actually return a body and we fail to serialize it - that's why the `HttpResponseException` is returned.

Reply:
Gotcha, that is an interesting thing to know, wonder if this is a gap of functionality in the de-serialization logic that we should close the gap on. If you have a quick repo on hand mind filing an issue on this.